conans/client/rest/uploader_downloader.py

```diff
@@ -141,6 +140,26 @@ def download(self, url, file_path=None, auth=None):
             raise ConanConnectionError("Download failed, check server, possibly try again\n%s"
                                        % str(e))
 
+    def _download_file(self, url, auth):
+        response = self.requester.get(url, stream=True, verify=self.verify, auth=auth)
+
+        if not response.ok:
+            raise ConanException("Error %d downloading file %s" % (response.status_code, url))
+
+        return response
+
+    def call_with_retry(self, retry, retry_wait, method, *args, **kwargs):
+        for counter in range(retry):
+            try:
+                return method(*args, **kwargs)
+            except Exception as exc:
```

Comment:
Too generic, I think we should repeat only in controlled exceptions (ConanException, or maybe beter ConanConnectionError)